gemini/function-calling/sql-talk-app/README.md

```diff
@@ -33,11 +33,24 @@ Try using the demo app now! [https://sql-talk-r5gdynozbq-uc.a.run.app/](https://
 4. Run the app with `streamlit run app.py`
 5. Navigate to the app in your browser at a URL similar to `http://localhost:8501`
 
+## Configuring a service account
+
+When deploying this app to Cloud Run, it is recommended that you [create a
+service account](https://cloud.google.com/iam/docs/service-accounts-create) to
+attach the following roles to, which are the permissions required for the app to
+read data from BigQuery, run BigQuery jobs, and use resources in Vertex AI:
+
+- [BigQuery Data Viewer](https://cloud.google.com/bigquery/docs/access-control#bigquery.dataViewer) (`roles/bigquery.dataViewer`)
+- [BigQuery Job User](https://cloud.google.com/bigquery/docs/access-control#bigquery.jobUser) (`roles/bigquery.jobUser`)
+- [Vertex AI User](https://cloud.google.com/vertex-ai/docs/general/access-control#aiplatform.user) (`roles/aiplatform.user`)
+
 ## Deploy the app to Cloud Run
 
-To deploy this app to Cloud Run, run the following command to have the app built
-with Cloud Build and deployed to Cloud Run, replacing the `service-account` and
-`project` values with your own values, similar to:
+To deploy this app to
+[Cloud Run](https://cloud.google.com/run/docs/deploying-source-code), run the
+following command to have the app built with Cloud Build and deployed to Cloud
+Run, replacing the `service-account` and `project` values with your own values,
+similar to:
 
 ```shell
 gcloud run deploy sql-talk --allow-unauthenticated --region us-central1 --service-account sql-talk@koverholt-devrel-355716.iam.gserviceaccount.com --source .
```

Comment:
Should you remove your own service account name?

Reply:
Yep will do. It's not sensitive info, but I might get questions if/when folks try to deploy that line verbatim!